src/main/java/nextstep/subway/domain/pathfinder/PathFinder.java

```diff
@@ -0,0 +1,50 @@
+package nextstep.subway.domain.pathfinder;
+
+import nextstep.subway.application.dto.PathResponse;
+import nextstep.subway.domain.CustomWeightedEdge;
+import nextstep.subway.domain.PathType;
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.Station;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+public abstract class PathFinder {
+    abstract public boolean isType(PathType pathType);
+    public PathResponse findPath(final List<Section> sections, final Station sourceStation, final Station targetStation) {
+        checkSameStation(sourceStation, targetStation);
+
+        WeightedMultigraph<Station, CustomWeightedEdge> graph = createGraph(sections);
+
+        checkStationContainsGraph(sourceStation, targetStation, graph);
+
+        DijkstraShortestPath<Station, CustomWeightedEdge> dijkstraShortestPath = new DijkstraShortestPath<>(graph);
+        final GraphPath<Station, CustomWeightedEdge> path = dijkstraShortestPath.getPath(sourceStation, targetStation);
+
+        return createPathResponse(path);
+    }
+
+    abstract protected WeightedMultigraph<Station, CustomWeightedEdge> createGraph(final List<Section> sections);
```

Comment:
[컨벤션](https://google.github.io/styleguide/javaguide.html#s4.8.7-modifiers)에 따르면 접근 제어가 최우선 입니다 😃 
```suggestion
    protected abstract WeightedMultigraph<Station, CustomWeightedEdge> createGraph(final List<Section> sections);
```